api/v1/common_types.go

```diff
@@ -282,6 +285,13 @@ const (
 	// DataflowStateInSync describes the status of a NifiDataflow as in sync
 	DataflowStateInSync DataflowState = "InSync"
 
+	// ParameterContextSecretStateUpToDate describes the status of NifiParameterContext secrets as up-to-date
```

Comment:
Is there a way to handle this generally for all places nifikop has secret refs? Specifically, can this be done without adding a state field?

Perhaps there's a way to detect dirtiness without storing the result in status. This way, we could use the same approach for every place we have secret refs across nifikop.

Thoughts?

Reply:
The problem is that updating the secret triggers a reconciliation. But the `NifiParamerContext` reconciliation is always in a loop. So the only way I've found to identify a reconciliation triggered by a secret update to a classic reconciliation is to use another controller to monitor the secrets and set a state in the resource to notify the other controller to update because the secrets have changed.

The other solution is to update the sensitive parameter directly in the new controller, but this will require a lot of duplicated code or a refactoring of the controller logic.

I haven't found a way to pass additional information in `reconcile.Request` to indicate the reason for the reconciliation.

Maybe you've thought of something I haven't?

Reply:
Yeah i agree that a controller dedicated to watch secrets is the way to go. Perhaps we could have this controller do more than just handle parameter contexts referring to secrets. We could create an internal API for handling secret updates for controllers that contain secret refs (e.g. parameter context, nifi cluster,  nifi user)? For example:

1. Use a controller dedicated to watching for secrets in the namespace(s) nifikop is told to watch.
2. When a secret gets updated, a reconciliation of this controller is triggered.
3. Determine which resource is referring to the updated secret. 
4. Call the internal API (i.e. `nifiCluster.handleSecretUpdate(updatedSecret)`) to take the appropriate action given the updated secret

This way, we wouldn't need a status field on the respective resources and we can just take an action when a secret gets updated.

Thoughts?

Reply:
I'm not about the last point. How do you see it?

Reply:
Any idea?

Reply:
My apologies on this. I don't want to hold up this PR. I think this works - my motivation for this thread was just to have a general purpose way of watching for secret updates that isn't specific to parameter contexts. 

That said, if we need a general purpose way we can always build one in the future.